solutions/azure-hub-spoke-connected-group/bicep/modules/avnm.bicep

```diff
@@ -0,0 +1,315 @@
+      }
+    }
+  }
+  resource networkGroupNonProd 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-spokes-nonprod'
+    properties: {
+      description: 'Non-prod Spoke VNETs Network Group'
+    }
+    resource staticMembersSpokeOne 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembersSpokeTwo 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[1]
+      }
+    }
+  }
+  resource networkGroupAll 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-all'
+    properties: {
+      description: 'All VNETs Network Group (for Security Configurations)'
+    }
+    resource staticMembers1 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-0'
+      properties: {
+        resourceId: prodNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembers2 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: prodNetworkGroupMembers[1]
+      }
+    }
+    resource staticMembers3 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembers4 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-3'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[1]
+      }
+    }
+    resource staticMembers5 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-4'
+      properties: {
+        resourceId: hubVnetId
+      }
+    }
+  }
+}
+
+@description('This connectivity configuration defines the connectivity between the spokes. Only deployed if requested.')
+resource connectivityConfigurationNonProd 'Microsoft.Network/networkManagers/connectivityConfigurations@2022-05-01' = {
+  name: 'cc-${location}-spokesnonprod'
+  parent: networkManager
+  dependsOn: [
+    networkManager::networkGroupNonProd::staticMembersSpokeOne
+    networkManager::networkGroupNonProd::staticMembersSpokeTwo
+  ]
+  properties: {
+    description: 'Non-prod poke-to-spoke connectivity configuration'
+    displayName: 'Non-prod Spoke-to-Spoke Connectivity'
+    appliesToGroups: [
+      {
+        networkGroupId: networkManager::networkGroupNonProd.id
+        isGlobal: 'False'
+        useHubGateway: string(deployVpnGateway)
+        groupConnectivity: 'DirectlyConnected'
+      }
+    ]
+    connectivityTopology: 'HubAndSpoke'
+    deleteExistingPeering: 'True'
+    hubs: [ 
+      {
+        resourceId: hubVnetId
+        resourceType: 'Microsoft.Network/virtualNetworks'
+      } 
+    ]
+    isGlobal: 'False'
+  }
+}
+
+@description('This connectivity configuration defines the connectivity between the spokes. Only deployed if requested.')
+resource connectivityConfigurationProd 'Microsoft.Network/networkManagers/connectivityConfigurations@2022-05-01' = {
```

Comment:
Comment here implies it's conditional at this point. But it doesn't appear to be.